test/sagemaker_tests/pytorch/training/conftest.py

```diff
@@ -563,6 +563,25 @@ def skip_p5_tests(instance_type, processor, ecr_image):
         if processor != "gpu" or Version(image_cuda_version.strip("cu")) < Version("120"):
             pytest.skip("P5 EC2 instance require CUDA 12.0 or higher.")
 
+@pytest.fixture(autouse=True)
+def skip_smdataparallel_p5_tests(instance_type, processor, ecr_image):
+    """SMDDP tests are broken for PyTorch 2.1 on p5 instances, so we should skip
+    """
+    smdataparallel_skip = False
+    skip_dict = {"==2.1": ["cu121"]}
```

Comment:
nit: I believe ==2.1 will resolve to ==2.1.0; if you want this to apply to all patch versions of 2.1, you may want to use ==2.1.*

Reply:
I think I've addressed this in the latest updates.